src/components/App.js

```diff
@@ -63,13 +63,13 @@ type Props = {
   endPanelCollapsed: boolean,
   activeSearch: string,
   quickOpenEnabled: boolean,
-  setActiveSearch: string => void,
-  closeActiveSearch: () => void,
-  closeProjectSearch: () => void,
-  openQuickOpen: (query?: string) => void,
-  closeQuickOpen: () => void,
-  setOrientation: OrientationType => void,
-  canRewind: boolean
+  canRewind: boolean,
+  setActiveSearch: typeof actions.setActiveSearch,
```

Comment:
In the profiler we use a lot of type inference for our actions like this. I've been a big fan of it from the maintainability of the code.

* There is only one type definition, the action creator.
* When modifying the action creator, all of the consuming components will provide actionable errors when the type signature changes.
* There are not duplicate definitions all over the place.

The one problem with this is that `mapDispatchToProps` actually modifies the type signature of the functions, and the react-redux definition does not take this into account. This can cause a few inconsistencies in approach.

Original type signatures of the action creator:

```
// Action creator
(Arg1, Arg2) => Action
// Thunk action creator
(Arg1, Arg2) => (Dispatch, GetState) => ReturnType
```

Then `mapDispatchToProps` changes them to:

```
// Action creator
(Arg1, Arg2) => void
// Thunk action creator
(Arg1, Arg2) => ReturnType
```

I'm not sure a great solution to this. It turns out some of the profiler's types aren't correct here either.